src/builder.ts

```diff
@@ -27,31 +28,74 @@ interface FieldPublisherConfig {
   ordering?: boolean | Record<string, boolean>
 }
 
+export interface Options {
+  photon?: (ctx: any) => any
+  shouldGenerateArtifacts?: boolean
+  inputs?: {
+    photon?: string
```

Comment:
I didn't pay attention to this addition (`inputs.photon`), but I'd say that sucks 😞
While it's a simple field, we're degrading the surface API to reduce our internal complexity and I don't like it

Not only are we leaking technical details to the surface API (`inputs.photon` is the path we need to require DMMF and users don't need to know about that), but we also end up with a weird duplicated API: `photon` and `inputs.photon`, which IMO doesn't make sense from a user perspective.

We could for sure enhance it by renaming `inputs.photon` to something like `inputs.photonPath`, but it'd still force users to input photon's path twice (once to import photon's instance to pass it to their graphql server context, and a second time to pass it to nexus-prisma) on the worst case scenario, and again, I think that's not very elegant

I'd suggest waiting before we make a decision on the DMMF generator. If we agree on the DMMF generator, we'd end up with a predictable location to require the DMMF and therefore remove the need for `inputs.photon`

Reply:
Agree it sucks, naming was an attempt to use symmetrical terms (i/o) and following from nexus api (outputs).

Do think its an ok tradeoff if it means they don't have to write a generator config block. There's a clear tradeoff there. And it seems acceptable.